src/views/blocks/rsvp/form/form.php

```diff
@@ -8,53 +8,56 @@
  *
  * See more documentation about our Blocks Editor templating system.
  *
- * @link    {INSERT_ARTICLE_LINK_HERE}
+ * @link    https://m.tri.be/1amp Help article for RSVP & Ticket template files.
  *
  * @since   4.9
  * @since   4.12.0 Add $post_id to filter for hiding opt-outs.
  * @since   4.12.3 Add comments to help IDE auto-completion. Array short syntax.
+ * @since   TBD Add docblock vars and use $ticket->ID instead of duplicative $ticket_id.
  *
- * @version 4.12.3
+ * @version TBD
+ *
+ * @var Tribe__Tickets__Editor__Template $this    Template object.
+ * @var int                              $post_id [Global] The current Post ID to which RSVPs are attached.
+ * @var Tribe__Tickets__Ticket_Object    $ticket  The ticket object with provider set to RSVP.
+ * @var string                           $going   The RSVP status at time of add/edit, or empty if not in that context.
  */
 
-$ticket_id = $this->get( 'ticket_id' );
-$post_id   = $this->get( 'post_id' );
-$going     = $this->get( 'going' );
-
 /** @var Tribe__Tickets__Tickets_Handler $tickets_handler */
 $tickets_handler = tribe( 'tickets.handler' );
-$ticket_data     = $tickets_handler->get_object_connections( $ticket_id );
+
+$ticket_data = $tickets_handler->get_object_connections( $ticket->ID );
 
 $event_id = $ticket_data->event;
 
 /** @var Tribe__Tickets__RSVP $rsvp */
-$rsvp       = tribe( 'tickets.rsvp' );
-$must_login = ! is_user_logged_in() && $rsvp->login_required();
+$rsvp = tribe( 'tickets.rsvp' );
+
+$must_login = $this->get( 'must_login' );
 ?>
 <form
 	name="tribe-rsvp-form"
-	data-product-id="<?php echo esc_attr( $ticket_id ); ?>"
+	data-product-id="<?php echo esc_attr( $ticket->ID ); ?>"
 >
-	<input type="hidden" name="product_id[]" value="<?php echo esc_attr( absint( $ticket_id ) ); ?>">
+	<input type="hidden" name="product_id[]" value="<?php echo esc_attr( absint( $ticket->ID ) ); ?>">
 	<input type="hidden" name="attendee[order_status]" value="<?php echo esc_attr( $going ); ?>">
 	<!-- Maybe add nonce over here? Try to leave templates as clean as possible -->
 
 	<div class="tribe-left">
 		<?php if ( ! $must_login ) : ?>
-			<?php $this->template( 'blocks/rsvp/form/quantity', [ 'ticket' => $ticket ] ); ?>
+			<?php $this->template( 'blocks/rsvp/form/quantity' ); ?>
 		<?php endif; ?>
 	</div>
 
 	<div class="tribe-right">
 		<?php $this->template( 'blocks/rsvp/form/error' ); ?>
 
 		<?php if ( $must_login ) : ?>
-			<?php $this->template( 'blocks/rsvp/form/submit-login', [ 'event_id' => $event_id, 'going' => $going, 'ticket_id' => $ticket_id ] ); ?>
+			<?php $this->template( 'blocks/rsvp/form/submit-login', [ 'event_id' => $event_id, ] ); ?>
```

Comment:
* :no_entry_sign: Comma not allowed after last value in single-line array declaration
* :no_entry_sign: Comma not allowed after last value in single-line array declaration

<sub>hash: 64220f7c49942418adead70e5e2d3bd3</sub>